### PR TITLE
Refactor core process extensions

### PR DESCRIPTION
**Pull request by sirh3e**
See the commits for more information

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=789) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=789) before we can accept your contribution.<br/><br/>:x: sirh3e [sign now](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=789)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/tye?pullRequest=789) it.</sub>

File: src/Microsoft.Tye.Core/ProcessExtensions.cs [hunk 1]
```diff
@@ -51,26 +51,24 @@ private static void GetAllChildIdsUnix(int parentId, ISet<int> children, TimeSpa
...
-                                children.Add(id);
-                                // Recursively get the children
-                                GetAllChildIdsUnix(id, children, timeout);
-                            }
-                        }
+                        return;
                     }
+
+                    if (!int.TryParse(text, out var id))
+                        continue;
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/ProcessExtensions.cs:67]: nit: prefer braces around ifs.

File: src/Microsoft.Tye.Core/ProcessExtensions.cs [hunk 2]
```diff
@@ -51,26 +51,24 @@ private static void GetAllChildIdsUnix(int parentId, ISet<int> children, TimeSpa
                     timeout,
                     out var stdout);
 
-                if (!string.IsNullOrEmpty(stdout))
+                if (string.IsNullOrEmpty(stdout))
+                    return;
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/ProcessExtensions.cs:55]: same here.

**Review by jkotalik — APPROVED** [review]

**Comment by sirh3e** [comment]
Added braces

**Pull request merged**